controller/src/main/java/org/jboss/as/controller/persistence/AbstractConfigurationPersister.java

```diff
@@ -86,13 +89,12 @@ public ModelNode getModelNode() {
 
                     @Override
                     public XMLElementWriter<SubsystemMarshallingContext> getSubsystemWriter(String extensionName) {
-                        synchronized (subsystemWriters) {
-                            return subsystemWriters.get(extensionName);
-                        }
+                        return subsystemWriters.computeIfAbsent(extensionName, name -> subsystemWriterSuppliers.getOrDefault(name, () -> null).get());//lazy create writer, but only once per config serialization
```

Comment:
subsystemWriters should now be a final local var in marshalAsXml. Then there is no need for the instance field any more and no need for the clear() call. If it's an instance var the "clear" call becomes a possible concurrency issue some day, e.g. concurrent CLI requests to read-config-as-xml.

```
final Map<String, XMLElementWriter<SubsystemMarshallingContext>> subsystemWriters = new HashMap<>();
ModelMarshallingContext extensibleModel = new ModelMarshallingContext() {
```